t/ui/16-tests_job_next_previous.t

```diff
@@ -276,5 +276,38 @@ subtest 'bug reference shown' => sub {
     );
 };
 
+subtest 'server-side limit has precedence over user-specified limit' => sub {
+    my $limits = OpenQA::App->singleton->config->{misc_limits};
+    $limits->{next_ajax_max_limit} = 5;    # set maximum limit for next
+    $limits->{previous_ajax_max_limit} = 5;    # set maximum limit for previous
+    $limits->{next_ajax_default_limit} = 2;    # set default limit for next
+    $limits->{previous_ajax_default_limit} = 2;    # set default limit for previous
+
+    $t->get_ok('/tests/99910/ajax?previous_limit=0&next_limit=6', 'query with exceeding user-specified limit for next')
+      ->status_is(200);
+    my $jobs = $t->tx->res->json->{data};
+    is ref $jobs, 'ARRAY', 'data returned (1)' and is scalar @$jobs, 7, 'maximum limit for next is effective';
```

Comment:
Tests are much easier to maintain long term when you put each `is` statement on a separate line.

Reply:
Here I think it makes sense to combine lines because only if the result is a proper array the second check makes sense, otherwise the error message could be confusing.

Reply:
A fair point. I would just leave out the first test though and have it die on the second one.